ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/authorization/AsyncForwardingListener.scala

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.server.api.authorization
+
+import io.grpc.ServerCall
+
+/** This listener buffers all messages until `setNextListener` is called,
+  * at which point all buffered messages are sent to the given listener.
+  * From then on, all future messages are sent directly to the given listener.
+  *
+  * The target listener is usually created through `Contexts.interceptCall` or `ServerCallHandler.startCall`.
+  * */
+abstract class AsyncForwardingListener[ReqT] extends ServerCall.Listener[ReqT] {
+  protected type Listener = ServerCall.Listener[ReqT]
+  private var stash: List[Listener => Unit] = List.empty
```

Comment:
Probably not particularly problematic, but it probably makes more sense to use a data structure with sub-linear appending complexity.

Reply:
What would be a good one? A mutable list? Or just prepend all messages and then reverse the list once before flushing it?

Reply:
I was using a `private[this] val stash: mutable.ListBuffer`